jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -115,6 +123,9 @@ def _data_selected_changed(self, event={}):
         # Auto-populate background, if applicable
         self._bg_subset_selected_changed()
 
+        # update self._selected_subset with the new self._selected_data
+        self._subset_selected_changed()
+

```

Comment:
I'm not 100% sure this is necessary, but just following the logic it seemed that it might be in some cases.  @pllim?

Reply:
I do not think it is necessary? I only trigger background because when image change, the background level changes. But the data subset is not used until calculation button is pressed.

Reply:
What about the scenario where the user changes `data_selected` _after_ `subset_selected`, since `_get_region_from_subset` appears to check for `self.data_selected` before returning and setting `self._selected_subset`?

Reply:
Good point. Right now, it does not matter because of glue-viz/glue-astronomy#52 but when that is fixed, you are probably right. Maybe you just fixed a very subtle bug. 😅

Reply:
The current logic crash if data is selected but there is no subset available. You need to modify it to not crash. To reproduce:

1. Open image in Imviz.
2. Open photometry plugin.
3. Select a Data.
4. A red snackbar pops up saying Subset something something failed.

Reply:
I don't see a crash in photometry, but rather the subset dropdown is empty and the rest of the plugin UI does not appear (I don't think this is different from the previous behavior).

I just "fixed" the case where you create and select a subset but then delete it.  This now defaults to an empty selection (and hides the rest of the plugin) when there is no "default entry", but otherwise sets the default entry ("Entire Spectrum" for a lot of the spectral subset dropdowns, or "Manual" for the background dropdown in photometry).

Reply:
Sorry, by crash, I mean a red Snackbar on the viewer. You didn't see it?

Reply:
Ok... I probably just missed it, but I think my last commit will have accidentally fixed it (it now returns without extracting if the dropdown is empty, but doesn't show the rest of the plugin still).